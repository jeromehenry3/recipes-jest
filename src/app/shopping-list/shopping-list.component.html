<div class="row">
  <div class="col-xs-10">
    <app-shopping-edit></app-shopping-edit>
    <hr />
    <p>ici vient la liste</p>
  </div>
</div>
