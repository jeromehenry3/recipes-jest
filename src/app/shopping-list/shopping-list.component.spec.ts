import { async, ComponentFixture, TestBed } from '@angular/core/testing';
import { MockComponent } from 'ng-mocks';

import { ShoppingListComponent } from './shopping-list.component';
import { ShoppingEditComponent } from './shopping-edit/shopping-edit.component';
import { By } from '@angular/platform-browser';

describe('ShoppingListComponent', () => {
  let component: ShoppingListComponent;
  let fixture: ComponentFixture<ShoppingListComponent>;

  beforeEach(async(() => {
    TestBed.configureTestingModule({
      declarations: [
        ShoppingListComponent,
        MockComponent(ShoppingEditComponent)
      ]
    })
    .compileComponents();
  }));

  beforeEach(() => {
    fixture = TestBed.createComponent(ShoppingListComponent);
    component = fixture.componentInstance;
    fixture.detectChanges();
  });

  it('should create', () => {
    expect(component).toBeTruthy();
  });

  it('should match snapshot', () => {
    const shoppingList = fixture.debugElement.nativeElement;
    expect(shoppingList).toMatchSnapshot();
  });

  it('should render a `ShoppingEditComponent`', () => {
    const shoppingEdit = fixture.debugElement.queryAll(By.directive(ShoppingEditComponent));
    expect(shoppingEdit).toHaveLength(1);
  });
});
