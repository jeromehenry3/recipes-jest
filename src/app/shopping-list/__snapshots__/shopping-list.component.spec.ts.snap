// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ShoppingListComponent should match snapshot 1`] = `
<div
  id="root1"
>
  <div
    class="row"
  >
    <div
      class="col-xs-10"
    >
      <app-shopping-edit />
      <hr />
      <p>
        ici vient la liste
      </p>
    </div>
  </div>
</div>
`;
